unix/x0vncserver/XDesktop.cxx

```diff
@@ -214,6 +214,15 @@ XDesktop::XDesktop(Display* dpy_, Geometry *geometry_)
   }
 #endif
 
+  // Need to set an error handler for XLib because most errors are reported
+  // asynchronously and default handler terminates the process on any error.
+  XSetErrorHandler([](Display* d, XErrorEvent* ev) -> int {
+    char buff[1024];
+    XGetErrorText(d, ev->error_code, buff, sizeof(buff));
+    vlog.error("XLib error: %s", buff);
+    return 0;
+  });
```

Comment:
This is an unrelated change IMO, and a controversial one. All the existing code is written with the default error handler in mind.

What issue was this meant to resolve?

Reply:
It was meant to avoid server crash due to possible XLib errors. During testing, I occasionally closed the app which had sent selection requests to x0vncserver, while server was paused in debugger. This would lead to `BadWindow` error once server was resumed and it would terminate. This is very unlikely to happen in real life, but I added the handler just in case.

I didn't realize this would be controversial, sorry about that. I will remove that commit.

Reply:
No worries. It's not expected that everyone keeps track of all of X11's evil quirks. :)